sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEShuffleReadExec.scala

```diff
@@ -144,12 +150,7 @@ case class AQEShuffleReadExec private(
     }
   }
 
-  private def shuffleStage = child match {
-    case stage: ShuffleQueryStageExec => Some(stage)
-    case _ => None
-  }
-
-  @transient private lazy val partitionDataSizes: Option[Seq[Long]] = {
+  private def partitionDataSizes: Option[Seq[Long]] = {
```

Comment:
only called once, not see a benefit to materialize it

Reply:
can we add `assert(shuffleStage.isDefined)` like what we did in `lazy val shuffleRdd`?